agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/ExampleExporter.java

```diff
@@ -0,0 +1,342 @@
+                    if(matchFound && attribute.key!=null) {
+                        //found a match
+                        if(existingSpanAttributes.get(attribute.key)!=null) {
+                            if(attribute.value!=null) {
+                                // found a match with value
+                                AttributeValue existingAttributeValue=existingSpanAttributes.get(attribute.key);
+                                if(existingAttributeValue.getType() == Type.STRING && existingAttributeValue.getStringValue().equals(attribute.value)) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                            } else {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            return matchFound;
+
+        } else if(excludes.match_type.equalsIgnoreCase("regexp")) {
+            boolean matchFound = false;
+            if(excludes.span_names!=null && excludes.span_names.size()>0) {
+                for(String service:excludes.span_names) {
+                    if(spanName.matches(service)) {
+                        matchFound = true;
+                        break;
+                    }
+                }
+            } else {
+                matchFound = true;
+            }
+
+            if(excludes.attributes!=null && excludes.attributes.size()>0) {
+                for(SpanProcessorAttribute attribute:excludes.attributes) {
+                    if(matchFound && attribute.key!=null) {
+                        //found a match
+                        if(existingSpanAttributes.get(attribute.key)!=null) {
+                            if(attribute.value!=null) {
+                                // found a match with value
+                                AttributeValue existingAttributeValue=existingSpanAttributes.get(attribute.key);
+                                if(existingAttributeValue.getType() == Type.STRING && existingAttributeValue.getStringValue().equals(attribute.value)) {
+                                    return true;
+                                } else {
+                                    return false;
+                                }
+                            } else {
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+
+            return matchFound;
+        }
+        return false;
+    }
+
+    private boolean checkIncludes(SpanProcessorIncludeExclude includes, ReadableAttributes existingSpanAttributes, String spanName) {
+        if(includes==null || includes.match_type==null) return true;
+        else if(includes.match_type.equalsIgnoreCase("strict")) {
+            if(includes.span_names!=null && includes.span_names.size()>0) {
```

Comment:
we can normalize spanNames in "optimized" model to never be null
```suggestion
            if(!includes.span_names.isEmpty()) {
```